File: api/src/test/java/net/blueshell/api/factory/dto/request/PasswordResetRequestFactory.java
package net.blueshell.api.factory.dto.request;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.recovery.PasswordResetRequest;
import net.blueshell.api.factory.dto.BaseDtoFactory;
import org.springframework.stereotype.Component;

/**
 * Factory for PasswordResetRequest test instances.
 */
@Component
@RequiredArgsConstructor
public class PasswordResetRequestFactory extends BaseDtoFactory<PasswordResetRequest> {

    @Override
    public Class<PasswordResetRequest> targetType() {
        return PasswordResetRequest.class;
    }

    @Override
    public PasswordResetRequest createBasic() {
        PasswordResetRequest dto = new PasswordResetRequest();
        dto.setToken(unique("tok"));
        dto.setPassword("Password123!");
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/request/PasswordResetRequestFactory.kt
package net.blueshell.api.factory.dto.request

import net.blueshell.api.auth.web.dto.recovery.PasswordResetRequest
import net.blueshell.api.factory.dto.BaseDtoFactory
import org.springframework.stereotype.Component

/**
 * Factory for PasswordResetRequest test instances.
 */
@Component
class PasswordResetRequestFactory : BaseDtoFactory<PasswordResetRequest>() {

    override fun targetType(): Class<PasswordResetRequest> = PasswordResetRequest::class.java

    override fun createBasic(): PasswordResetRequest {
        val dto = PasswordResetRequest()
        dto.token = unique("tok")
        dto.password = "Password123!"
        return dto
    }
}
